.github/workflows/test.yml

```
@@ -0,0 +1,29 @@
+name: Ruby CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

```suggestion
  pull_request:
    types: [opened, synchronize, reopened]
```

If we added that, we'd run the job for all commits, for all PRs.

I searched around, and found this longer explanation of those values:
https://github.community/t/trigger-workflow-for-a-pull-request-and-for-all-subsequent-commits-in-it/17996

---

I ended up merging this down to the default branch in my fork because I couldn't figure out if I could run tests from a branch if I was just creating a `.workflows` for the first time. I think I was wrong about that being necessary, but more importantly...

Is this PR being based on my default branch preventing y'all from making any changes? There's a checkbox that does say it should allow edits...but I'm not sure if that actually works if I'm using the default branch in my fork. Apologies if that's the case and I'm happy to resubmit from a branch if that would help y'all.

---

We just have limited bandwidth.

---

Totally understand, no rush on my part, just wanted to make sure I wouldn't be a bottleneck.

I should say in the spirit of transparency that in between opening this PR and now, I started working at GitHub. I don't work on anything related to Actions and don't think there's any conflict between working there and helping this PR land, but wanted to let y'all know.

---

With respect to types, my reading of https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request is that the types you've listed are the default, but it's not bad to be explicit.

I also think it's not a bad idea to remove the limit of only running for PRs where the master branch is part of it, all PRs deserver CI 😄